dotnet-sdk-80/snapcraft.yaml

```diff
@@ -47,6 +47,54 @@ parts:
         craftctl set version="${DOTNET_SDK_VERSION}"
       fi
 
+  mounts:
+    plugin: dump
+    source: .
+    after:
+      - dotnet-sdk
+    override-build: |
+      RUNTIME_VERSION=$(ls ${CRAFT_STAGE}/usr/lib/dotnet/shared/Microsoft.NETCore.App/ | grep 8.0)
+      SDK_VERSION=$(ls ${CRAFT_STAGE}/usr/lib/dotnet/sdk | grep 8.0)
+      
+      if [ "${CRAFT_ARCH_BUILD_FOR}" = "amd64" ]; then
+        ARCH="x64"
+      elif [ "${CRAFT_ARCH_BUILD_FOR}" = "arm64" ]; then
+        ARCH="arm64"
+      else
+        echo "Unknown architecture (${CRAFT_ARCH_BUILD_FOR})"
+        exit 1
+      fi
+
+      for file in mounts/*; do
+        sed -i "s/{RT_VERSION}/$RUNTIME_VERSION/g" $file
+        sed -i "s/{SDK_VERSION}/$SDK_VERSION/g" $file
+        sed -i "s/{SNAP}/$CRAFT_PROJECT_NAME/g" $file
+        sed -i "s/{ARCH}/$ARCH/g" $file
+
+        if [[ "$file" == *"{RT_VERSION}"* ]]; then
+          new_file=$(echo "$file" | sed "s/{RT_VERSION}/$RUNTIME_VERSION/")
+          mv "$file" "$new_file"
+          file=$new_file
+          echo "Renamed '$file' to '$new_file'"
+        fi
+        if [[ "$file" == *"{SDK_VERSION}"* ]]; then
+          new_file=$(echo "$file" | sed "s/{SDK_VERSION}/$SDK_VERSION/")
+          mv "$file" "$new_file"
+          file=$new_file
+          echo "Renamed '$file' to '$new_file'"
```

Comment:
issue: `$file` gets reassigned with a new value, before the old value gets logged

Reply:
Yep, good catch. This got fixed with the change from you suggestion above.